include/linker/common-ram.ld

```diff
@@ -8,6 +8,12 @@
 #if defined(CONFIG_GEN_ISR_TABLES) && defined(CONFIG_DYNAMIC_INTERRUPTS)
 	SECTION_DATA_PROLOGUE(sw_isr_table,,)
 	{
+		/*
+		 * Some arch requires an entry to be alined to 8 bytes
+		 * boundary to use double word load instruction.  See
+		 * include/sw_isr_table.h.
+		 */
+		. = ALIGN(8);
```

Comment:
A waste of four bytes isn't a big deal, but if the only architecture with this limitation is SPARC, we probably shouldn't be putting this into the global linker script like this.

Maybe define a Z_ARCH_SW_ISR_TABLE_ALIGN macro in the arch files with a default value of "." or something?

Reply:
I thought 4 bytes will justified for yet another macro and complexity, but sure, I think we can do that.
Which include file do you want me to add? Our linker scripts don't usually have much files included.  Many has only the following four files.

- autoconf.h
- linker/sections.h
- linker/linker-defs.h
- linker/linker-tool.h

I don't think we want to introduce another include file for this?

Reply:
The problem I see is that Z_ARCH_SW_ISR_TABLE_ALIGN needs to go to arch.h and that header is not inlcuded by the linker; we need some headers' rework for that, doesn't look so simple.

Another option is to add this as a CONFIG_ define via Kconfig, e.g. in arch/"ARCH"/Kconfig

Reply:
Right. CONFIG_ is much simpler but should I go with that option?

Reply:
Yes, a Kconfig option is not too bad